migration/migration-service/src/main/java/org/wso2/carbon/ei/migration/internal/EIMigrationServiceComponent.java

```diff
@@ -21,51 +21,78 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
 import org.wso2.carbon.ei.migration.migrate.SecureVaultPasswordMigrationClient;
+import org.wso2.carbon.ei.migration.migrate.RegistryTaskDeletionClient;
 import org.wso2.carbon.registry.core.service.RegistryService;
 
-/**
- * @scr.component name="org.wso2.carbon.ei.migration.client" immediate="true"
- * @scr.reference name="registry.service" interface="org.wso2.carbon.registry.core.service.RegistryService"
- * cardinality="1..1" policy="dynamic"  bind="setRegistryService" unbind="unsetRegistryService"
- */
+@Component(
+        name = "org.wso2.carbon.ei.migration.client",
+        immediate = true)
 public class EIMigrationServiceComponent {
 
     private static final Log log = LogFactory.getLog(EIMigrationServiceComponent.class);
 
-    /**
-     * Method to activate bundle.
-     *
-     * @param context OSGi component context.
-     */
+    @Activate
     protected void activate(ComponentContext context) {
-        if (Boolean.valueOf(System.getProperty("migrate"))) {
-            log.info("WSO2 EI migration bundle is activated");
-            log.info("Initiating WSO2 EI migration");
-            try {
-                SecureVaultPasswordMigrationClient migrateSecureVaultPasswords = new SecureVaultPasswordMigrationClient();
-                migrateSecureVaultPasswords.migratePasswords();
-                log.info("Successfully completed password migration");
-            } catch (Throwable e) {
-                log.error("Error occurred during migration", e);
-            }
+        log.info("WSO2 EI migration bundle is activated");
+        if (Boolean.valueOf(System.getProperty("migrate.esb"))) {
+            migrateSecureVaultPasswords();
+            deleteScheduledMPRegistryTasks();
+        } else if (Boolean.valueOf(System.getProperty("migrate.ei650"))) {
```

Comment:
shouldn't the property name be something like "migrated_version" where the value is either "esb" or "ei650"?

Reply:
Fixed